Mathlib/Algebra/Hom/GroupAction.lean

```diff
@@ -61,143 +144,286 @@ structure MulActionHom where
   /-- The underlying function. -/
   toFun : X → Y
   /-- The proposition that the function preserves the action. -/
-  map_smul' : ∀ (m : M') (x : X), toFun (m • x) = m • toFun x
-#align mul_action_hom MulActionHom
+  map_smul' : ∀ (m : M) (x : X), toFun (m • x) = (φ m) • toFun x
 
 /- Porting note: local notation given a name, conflict with Algebra.Hom.GroupAction
  see https://github.com/leanprover/lean4/issues/2000 -/
 @[inherit_doc]
-notation:25 (name := «MulActionHomLocal≺») X " →[" M:25 "] " Y:0 => MulActionHom M X Y
+notation:25 (name := «MulActionHomLocal≺») X " →ₑ[" φ:25 "] " Y:0 => MulActionHom φ X Y
+
+/-
+/-- Equivariant functions -/
+abbrev MulActionHom (M : Type _) (X Y : Type _) [SMul M X] [SMul M Y] := MulActionHom (@id M) X Y
+-/
+
+@[inherit_doc]
+notation:25 (name := «MulActionHomIdLocal≺») X " →[" M:25 "] " Y:0 => MulActionHom (@id M) X Y
 
-/-- `SMulHomClass F M X Y` states that `F` is a type of morphisms preserving
-scalar multiplication by `M`.
+/-- `MulActionHomClass F φ X Y` states that `F` is a type of morphisms which are `φ`-equivariant
 
 You should extend this class when you extend `MulActionHom`. -/
-class SMulHomClass (F : Type _) (M X Y : outParam <| Type _) [SMul M X] [SMul M Y] extends
+class MulActionHomClass (F : Type _) {M N : outParam (Type _)}
+(φ : outParam (M → N)) (X Y : outParam (Type _)) [SMul M X] [SMul N Y] extends
   FunLike F X fun _ => Y where
   /-- The proposition that the function preserves the action. -/
-  map_smul : ∀ (f : F) (c : M) (x : X), f (c • x) = c • f x
-#align smul_hom_class SMulHomClass
+  map_smul : ∀ (f : F) (c : M) (x : X), f (c • x) = (φ c) • (f x)
```

Comment:
[As discussed on Zulip](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.236057.20-.20Add.20equivariant.20morphisms.20and.20adjust.20linear.20maps) I'd reserve `map_smul` for the "true" preserving case. So for the class we'd write:
```suggestion
  map_smulₑ : ∀ (f : F) (c : M) (x : X), f (c • x) = (φ c) • (f x)
```

Reply:
It's `map_smul'` now, because it has to coincide with what's on LinearMap…